### DISCUSSION TITLE
[WIP] Icon Design Refresh

### DISCUSSION THREAD

**Discussion by yw4z** [OrcaSlicer/OrcaSlicer#4348] [Dev]
*** Important - Check my shared repo for latest designs first

I will constantly edit this post and will open a PR for them everyone agrees on best
You can track all changes from here
[https://github.com/yw4z/OrcaSlicer/tree/icon-designs/resources/images](https://github.com/yw4z/OrcaSlicer/tree/icon-designs/resources/images)
""DONT merge this to main** icon documents is not at final stage. i will convert them before sending PRs for final documents
To preview changes you can replace images on your resources/images folder. i'm making many minor changes and its hard to create changelog for each of them.
please don't hesitate to share any ideas

### OVERALL GOALS
• Make UI to easier to use with better presented icons
• Improving UI consistency
• Make them unique to Orca Slicer
• Making all icons to pixel perfect

Extras
• Updating all icons will free 2-3MB storage form setup file

### TITLE BAR ICONS
![Screenshot-20240303052426](https://github.com/SoftFever/OrcaSlicer/assets/28517890/320afc5b-aa75-4ec7-b97f-e77937fdcaae)

![Screenshot-20240303052034](https://github.com/SoftFever/OrcaSlicer/assets/28517890/d93c77e3-4e29-40f7-9b01-2875b22804fc)

• All icons pixel perfect now
• Used file icon for file menu
• Used hamburger icon for second menu
• Used line based icons for save, undo, redo
• Changed calibration icon because old design is looks small compared to new ones but tried to keep similar to old one • Changed minimize, maximize, restore, close buttons with simpler ones
• Total size for new icons reduced 7.5kb to 5.2kb
• Again icon resolutions was set wrong as 15x15px. corrected all as 18x18px. this fixed blurry icons • Aligned all icons to highlight box on hover
• there is also topbar_store, topbar_publish, topbar_open  is exist for titlebar images but i dont know how to see them in titlebar. you can remove these files if that features is already removed. i can rework them too if they can show with different config

### TAB BAR ICONS

![Screenshot-20240303022304](https://github.com/SoftFever/OrcaSlicer/assets/28517890/f93a1370-f18b-45b7-9e91-b13b2ec815f3)
![Screenshot-20240303021502](https://github.com/SoftFever/OrcaSlicer/assets/28517890/4c7ee793-1cb1-4efe-bb8a-9f22d78dd5b5)

• Home icon uses same line based design with other icons
• All icons reworked and pixel perfect now. so it will be render nicely on all resolutions
• All icon resolutions changed to 18x18 to 20x20 for perfect fit. Previous ones looks blurry due to wrong resolution
• Changed device icon with simpler printer icon. old icon has 1 screen and 1 printer but both of them so small and hard to identify 
• All icons vertically aligned to text. old version has 1px extra at bottom
• Total size for new icons reduced 6kb to 2.5kb
• Also tab_presets_inactive and tab_presets_active is exist for tab images but i dont know how to see them in tabs. you can remove these files if that feature is already removed

### TOOLBAR ICONS
![Screenshot-20240304041606](https://github.com/SoftFever/OrcaSlicer/assets/28517890/b01f5a6d-0b5b-44fb-92e3-28c198cdff9e)

![Screenshot-20240304023526](https://github.com/SoftFever/OrcaSlicer/assets/28517890/e5c94c15-c6be-4653-a79f-7a0aefcdf98c)

• All icons pixel perfect now
• Some icons uses 40x40 instead 39x40. That caused blurry images
• All icons redesigned or tweaked to fit to square. all visible as equal size now
• I tried to keep consistency between icons with using same arrow shapes, line width, proportions

• Reworked "Add Object". Resized cube to fill icon frame. Used isometric cube for better visual
• Reworked "Add Plate".Resized bed sheet to fill icon frame. Changed badge to fit nicely with icon frame
• Reworked "Auto Orient" and "Lay on Surface". It highlights object's edge/surface instead of all object. Added an * for auto because user cannot read auto text on that
• Designed new icons for "Split to Objects" and "Split to Parts". old ones just used letters and they are not representing functions
• Designed new icons for "Move, "Rotate", "Scale". Used same object at their center to increase similarity between them. Tried to increase similarity because all transform functions
• "Cut","Emboss". Simplified
• Measure > Added arrows to fill icon frame so it will look same size with other icons
• Assembly > Merged puzzle parts to increase size of icon to matching size of icon with others
• Seam Painting > Used same isometric cube on "Add Object" to improve consistency between icons
• Total size for changed icons reduced 92,6KB to 34,9kb

**ALTERNATIVES**
Alternative with caliper icon for measure function
![Screenshot-20240305030206](https://github.com/SoftFever/OrcaSlicer/assets/28517890/2aeeaf66-0cca-44a5-a1bb-97b10ff53623)

Alternatives for Seam Painting and Support Painting functions. i liked spray one
![Screenshot-20240304210900](https://github.com/SoftFever/OrcaSlicer/assets/28517890/62d15915-3c39-4084-a3eb-584cddc1ee81) 

### PLATE ICONS

![Screenshot-20240305194946](https://github.com/SoftFever/OrcaSlicer/assets/28517890/891b4aab-267e-48bb-8bba-e9c372f1607a)

![Screenshot-20240305195124](https://github.com/SoftFever/OrcaSlicer/assets/28517890/218d0dda-8fd6-4dd1-ade2-1492b9d3b748)

• Used a thinner border on icons for modern look
• Lock button changes color while locked. was only icon changes color before. this method will make identifying locked plates easier
• Applied same method for bed settings button

### ICONS ON UI COMPONENTS
--

### SIDEBAR ICONS
![Screenshot-20240305195640](https://github.com/SoftFever/OrcaSlicer/assets/28517890/d7d3a419-5834-4831-bd41-a6862106b3be)

• Updated all icons ot titler bars
![Screenshot-20240305201348](https://github.com/SoftFever/OrcaSlicer/assets/28517890/f4e255fe-74cc-44db-8a27-aa797269add6)
• Used spool icon on + and - too. I couldnt find how to use multi material feature when i started to use orca. This will guide user
• Previous compare icon was not presented feature nicely, i used A/B term on icon and made it much more simpler
• Used same printer icon tabs
• Used a fader / sliders on settings button for more modern look
• Aligned and resized  save / remove / search icons. they look equal size now
• Updated all icons on preset settings. ofc it needs more work
![Screenshot-20240305200428](https://github.com/SoftFever/OrcaSlicer/assets/28517890/6030158d-2969-4911-adca-299419cc6f51)

### PLACEHOLDER ICONS

![Screenshot-20240305200627](https://github.com/SoftFever/OrcaSlicer/assets/28517890/40113801-4f6a-4f60-a156-dc196126f259)

### EMBOSS POPUP ICONS
![Screenshot-20240305200847](https://github.com/SoftFever/OrcaSlicer/assets/28517890/670a2ea4-4665-49d3-bd56-848ae2c813f9)

• Simplified icons as much as possible
• Used color change for bold / italic buttons. it turns to cyan color if enabled

comparison New vs Old
![Screenshot-20240305202922](https://github.com/SoftFever/OrcaSlicer/assets/28517890/07128564-efae-4a25-ae00-7db7f46f3774)

### MODIFIER / OBJECT LIST ICONS / CONTEXT MENU

![Screenshot-20240306023545](https://github.com/SoftFever/OrcaSlicer/assets/28517890/d36b4d0e-5975-4d4c-b808-d7209a100ea2)

### MENUBAR ICONS

![Screenshot-20240306023216](https://github.com/SoftFever/OrcaSlicer/assets/28517890/6e390b5c-846f-40d3-b2a6-cb5f13714db9)

### VARIOUS ICONS

**LAYER VIEW MODE BUTTON**
![Screenshot-20240305105714](https://github.com/SoftFever/OrcaSlicer/assets/28517890/f7a18f90-ec0c-4fe8-b11b-f604f6061fab)

comparison of size change

![Screenshot-20240305105429](https://github.com/SoftFever/OrcaSlicer/assets/28517890/b0a5b73c-8177-45a9-a556-72a38acd3879)

• Size of icon increased around %150
• Removed size change on hovered images. used color change instead
• Total size for changed icons reduced 34,2KB to 13,1kb

### RELATED / FOUND BUGS

I will add pictures for all of them

**Bugs / missing features**
• Height range icon not shown on context menu and object list
• Process > Objects > Object list > not changes color of modifier icons. if icon is black its not possible to see on dark mode 

**UI inconsistencies**
• Process > Objects > Edit in parameter table > enable support > revert icon uses "lock_normal.svg" instead of "revert_btn.svg"
• Process > Objects > Object list > revert icon uses "lock_normal.svg" instead of "revert_btn.svg"
• Process > Objects > Edit in parameter table > value / input boxes not not visible until click
• Process > Objects > Search close button not working
• Process > Global > Search > "delete this preset" button still visible while search is active
• Clicking other toolbar button not closes variable layer height popup. also clicking outside not closing popup
• Toolbar icons blurry [4328](https://github.com/SoftFever/OrcaSlicer/issues/4328)
• Toolbar > Emboss > Height and depth not uses step value box. they use + / - buttons but other parts of UI uses step boxes
• Toolbar > Emboss > Advanced > Alignment buttons blurry
• Toolbar > Emboss > Style > Buttons not rendered on correct resolution (16x16). this causes blurry icons
• Toolbar > Emboss > Radio buttons for join cut modifier not uses general style for radio buttons
• Toolbar > Cut > Revert button always visible. It will only show up if value changed on other parts of UI

**Color related inconsistencies/ bugs**
• Title bar buttons not uses cyan background while hover
• Toolbar > Emboss > Advanced expansion button not uses cyan background while hover
• Toolbar > Emboss > Advanced > Sliders not uses cyan background for thumb
• Toolbar > Support Painting / Seam painting > Brush type selection radio button not uses cyan background for active button
• Printer Network Connection > Test icon rendered with wrong color if button disabled
• Context menu > Split > not changed color of icons. if icon is black its not possible to see on dark mode. they are white on toolbar. its ok to remove icons instead of fixing it
• Context menu > Simplify Model > Color of Apply button is not cyan while hovering. is green (bamboo studio color)
• Multiple plate slice > All bed stats text color is green (bamboo studio color)
• Filament settings > multi material > ramming settings > button not styled with dark theme
• Filament settings > multi material > ramming settings > popup window not works with dark theme
• Compare presets row highlighting is making too much contrast. and it makes hard to read rows
• Window resize icon right bottom is not compitable with dark theme

### REQUESTS / SUGGESTIONS
• There is no separate icons for  "Walls and Surfaces", Bridging, Overhangs, Overhang Speed, Jerk, on print settings. I can create icons for them if new image defined on code side
• Noticed this order would be better for toolbar icons
![Screenshot-20240305031523](https://github.com/SoftFever/OrcaSlicer/assets/28517890/37c399de-1e7b-4d93-b236-0cb30442556a)
• Toolbar > Scale > Not uses world coordinates term for XYZ axes. Also removing world coordinates text from Move and Rotate will make their popups smaller
• Sidebar > Process > Replace advance text with new atomic (advanced) icon

### HELP NEEDED
couldn't find or test these
• add.svg
• automatic_material_renewal.svg
• bind_machine.svg
• color_picker_border
• completed.svg
• copy_menu.svg
• flush_volumes.svg
• go_last_plate.svg
• go_next_plate.svg
• im_gcode_custom.svg
• im_gcode_pause.svg
• im_slider_delete.svg
• notification_eject_sd
• notification_preferences.svg
• notification_right.svg
• notification_documentation.svg
• open.svg
• open_in_browser.svg
• plate_settings_arrow.svg
• toolbar_arrow.svg
• toolbar_settings.svg
• unbind_machine.svg
• unbind.svg

There is more for this list. i will add overtime
Let me know if you see where they appear. Screenshots would be helpful with dark and light theme

**Comment by yw4z** [comment]
@SoftFever hi
you can check this repo for all changes
[https://github.com/yw4z/OrcaSlicer/tree/icon-designs/resources/images](https://github.com/yw4z/OrcaSlicer/tree/icon-designs/resources/images)
there are many changes on icons. most of icons changed except notification icons
i added wider tab icons on latest commit on that repo, here is a preview of it

![Screenshot-20240305193549](https://github.com/SoftFever/OrcaSlicer/assets/28517890/d44085ce-2001-4d72-bfe6-d4fc02c5091d)
also used a bigger toolhead for printer icon

**Comment by Eldenroot** [comment]
Nice!

**Comment by andrewcdash** [comment]
I'm a bit late to the party but did you still need help with any of this?

**Reply by yw4z** [reply]
like what? its already merged to current branch and many additional improvements has been made. its about 2 years old topic :)

**Reply by yw4z** [reply]
let me know if you think there are areas on UI can be improvement. i have a long list about them but changes requires a lot time

**Comment by andrewcdash** [comment]
Apologies. I just saw at the bottom of the page a section asking for help.
